pkg/controller/kmeshsecurity/security_manager.go

```diff
@@ -0,0 +1,201 @@
+	*pq = append(*pq, item)
+}
+
+func (pq *PriorityQueue) Pop() interface{} {
+	old := *pq
+	n := len(old)
+	item := old[n-1]
+	*pq = old[0 : n-1]
+	return item
+}
+
+var certs_maps sync.Map
+var log = logger.NewLoggerField("kmeshsecurity")
+
+type SecretManagerClient struct {
+	caClient *CitadelClient
+
+	// configOptions includes all configurable params for the cache.
+	configOptions *security.Options
+
+	// storing certificates
+	secretcache *sync.Map
+
+	caRootPath string
+
+	pending	PriorityQueue
+}
+ 
+var ScClient SecretManagerClient
+
+const (
+	rootCertPath       = "/var/run/secrets/istio/root-cert.pem"
+)
+   
+// NewSecretManagerClient creates a new SecretManagerClient.
+func NewSecretManagerClient() (*SecretManagerClient, error) {
+
+	tlsOpts = &TLSOptions{
+		RootCert:      rootCertPath,
+	}
+
+	options:= NewSecurityOptions()
+	caClient, err := NewCitadelClient(options, tlsOpts)
+	if err != nil {
+		return nil, err
+	}
+	pq := make(PriorityQueue, 0)
+	heap.Init(&pq)
+
+	ret := &SecretManagerClient{
+		caClient:      caClient,
+		configOptions: options,
+		caRootPath:  options.CARootPath,
+		secretcache: &certs_maps,
+		pending:	pq,
+	}
+	ScClient = *ret
+	go ScClient.delayedTask()
+	go ScClient.update_certs_routine()
+	go ScClient.delete_certs_routine()
+	return ret, nil
+}
+ 
+func GetSecretManagerClient() *SecretManagerClient {
+	return &ScClient
+}
+	
+// Automatically check and refresh when the validity period expires
+func (sc *SecretManagerClient) delayedTask() {
+	var new_certs *security.SecretItem
+	var err error
+	var expireTime time.Time
+
+	for {
+		expireTime = time.Now().Add(900 * time.Second)
+		if (len(sc.pending) > 0) {
+			next_tmp := heap.Pop(&sc.pending).(*Certs)
+			expireTime = next_tmp.exp
+			heap.Push(&sc.pending, next_tmp)
+		}
+		select{
+		case <-time.After(time.Until(expireTime.Add(-300 * time.Second))):
+			next := heap.Pop(&sc.pending).(*Certs)
+			if _, ok := sc.secretcache.Load(next.uid); !ok {
+				continue
+			} else {
+				new_certs, err = sc.caClient.fetch_cert(next.uid);
+				if err != nil {
+					<-time.After(10*time.Second)
+					new_certs, err = sc.caClient.fetch_cert(next.uid);
+				}
+			}
+			// Check if the key exists in the workload, if it does, refresh it, otherwise abandon it. 
+			// If multi-threading refresh errors occur in this round, the certificate will be deleted 
+			// in the next round of workload checks
+			_, ok := sc.secretcache.Load(next.uid);
+			if ok {
+				sc.secretcache.Store(next.uid, *new_certs)
+				heap.Push(&sc.pending, &Certs{exp: new_certs.ExpireTime, uid: next.uid})
+			}
+		case <-time.After(300 * time.Second):
+			continue
+		}
+		
+	}
+}
+
+// Initialize the certificate for the first time
+func (sc *SecretManagerClient) update_certs_routine() {
+	var new_certs *security.SecretItem
+	var err error
+	for workloadUid := range workload.SecurityAddDataChannel {
+		log.Infof("workloadUid: %v", workloadUid)
+		if certs, ok := sc.secretcache.Load(workloadUid); ok {
```

Comment:
```suggestion
		if cert, ok := sc.secretcache.Load(workloadUid); ok {
```
It is recommended that all subsequent `certs` be changed to` cert` if they refer to a cert.